Core/Core/TextSubmission/TextSubmissionViewController.swift

```diff
@@ -0,0 +1,153 @@
+        let controller = loadFromStoryboard()
+        controller.courseID = courseID
+        controller.editor = RichContentEditorViewController.create(env: .shared, context: .course(courseID), uploadTo: .myFiles)
+        controller.htmlContent = htmlContent
+        return controller
+    }
+
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .backgroundLightest
+        title = String(localized: "Text Entry", bundle: .core)
+
+        navigationController?.navigationBar.useModalStyle()
+        navigationItem.rightBarButtonItem?.accessibilityIdentifier = "TextSubmission.submitButton"
+        navigationItem.rightBarButtonItem?.isEnabled = false
+        addNavBarButtons()
+        editor.delegate = self
+        editor.placeholder = String(localized: "Write...", bundle: .core, comment: "Text submission editor placeholder")
+        editor.a11yLabel = String(localized: "Submission text", bundle: .core, comment: "Text submission editor accessibility label")
+        editor.webView.scrollView.layer.masksToBounds = false
+        embed(editor, in: contentView)
+
+        if htmlContent.isNotEmpty, isHorizonApp {
+            editor.setHTML(htmlContent)
+        }
+    }
+
+    public override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        keyboard = KeyboardTransitioning(view: view, space: keyboardSpace)
+    }
+
+    public func rce(_ editor: RichContentEditorViewController, canSubmit: Bool) {
+        navigationItem.rightBarButtonItem?.isEnabled = canSubmit
+    }
+
+    public func rce(_ editor: RichContentEditorViewController, didError error: Error) {
+        showError(error)
+    }
+
+    private func addNavBarButtons() {
+        if isHorizonApp {
+            navigationItem.rightBarButtonItem = UIBarButtonItem(
+                title: String(localized: "Done", bundle: .core),
+                style: .plain,
+                target: self,
+                action: #selector(sendHTMLContent)
+            )
+            navigationItem.leftBarButtonItem = UIBarButtonItem(
+                title: String(localized: "Cancel", bundle: .core),
+                style: .plain,
+                target: self,
+                action: #selector(cancel)
+            )
+        } else {
+            navigationItem.rightBarButtonItem = UIBarButtonItem(
+                title: String(localized: "Submit", bundle: .core),
+                style: .plain,
+                target: self,
+                action: #selector(submit)
+            )
+            addCancelButton(side: .left)
+        }
+    }
+
+    @objc private func sendHTMLContent() {
+        editor.getHTML { [weak self] html in
+            guard let self else {
+                return
+            }
+            didSetHtmlContent(html)
+            env.router.dismiss(self)
+        }
+    }
+
+    @objc private func cancel() {
+        didSetHtmlContent(htmlContent)
+        env.router.dismiss(self)
```

Comment:
Please either rename this method to indicate this cancel is only intended for the horizon logic, or use this method in the non-horizon variant as well